wasp/apps/alarm.py

```diff
@@ -86,21 +87,29 @@ def __init__(self):
         self.alarms = (bytearray(3), bytearray(3), bytearray(3), bytearray(3))
         self.pending_alarms = array.array('d', [0.0, 0.0, 0.0, 0.0])
 
-        # Set a nice default
-        self.num_alarms = 1
-        for alarm in self.alarms:
-            alarm[_HOUR_IDX] = 8
-            alarm[_MIN_IDX] = 0
-            alarm[_ENABLED_IDX] = 0
-        self.alarms[0][_ENABLED_IDX] = _WEEKDAYS
-
+        self.num_alarms = 0
+        try:
+            with open("alarms.txt", "r") as f:
+                alarms = f.readlines()[0].split(";")
+            if "" in alarms:
+                alarms.remove("")
+            for alarm in alarms:
+                n = self.num_alarms
+                h, m, st = map(int, alarm.split(","))
+                self.alarms[n][0] = h
+                self.alarms[n][1] = m
+                self.alarms[n][2] = st
+                self.num_alarms += 1
+        except Exception:
+            pass
+        self._set_pending_alarms()
 
     def foreground(self):
         """Activate the application."""
 
         self.del_alarm_btn = widgets.Button(170, 204, 70, 35, 'DEL')
         self.hours_wid = widgets.Spinner(50, 30, 0, 23, 2)
-        self.min_wid = widgets.Spinner(130, 30, 0, 59, 2)
+        self.min_wid = widgets.Spinner(130, 30, 0, 59, 2, 5)
```

Comment:
Subjective I know, but I liked the minute precision...

Reply:
While it's true that precise control over the alarm is mostly useful with the timer app, sometimes I want to be alerted on absolute --not relative-- time, and except for wake up alarms a 5 min precision is a bit rough. Imagine setting an alarm 5 mins before the bus arrives for example (so you can go to the stop). With 5 mins precision the interval between the alarm and when the bus arrives can go up to 9 mins.

Ideally I think we could have a way for the user to control the speed of spinners, like a >> button above the > button, or detection of fast taps or long presses, like most alarm clocks. But that's a bigger feature request.

Reply:
I think it would be better to add long press support : https://github.com/daniel-thompson/wasp-os/issues/331